osu.Game.Tests/Beatmaps/Formats/LegacyStoryboardDecoderTest.cs

```diff
@@ -95,6 +95,27 @@ public void TestDecodeStoryboardEvents()
             }
         }
 
+        [Test]
+        public void TestLoopWithoutExplicitFadeOut()
+        {
+            var decoder = new LegacyStoryboardDecoder();
+
+            using (var resStream = TestResources.OpenResource("animation-loop-no-explicit-end-time.osb"))
+            using (var stream = new LineBufferedReader(resStream))
+            {
+                var storyboard = decoder.Decode(stream);
+
+                StoryboardLayer background = storyboard.Layers.Single(l => l.Depth == 3);
+                Assert.AreEqual(1, background.Elements.Count);
+
+                Assert.AreEqual(2000, background.Elements[0].StartTime);
+                Assert.AreEqual(2000, (background.Elements[0] as StoryboardAnimation)?.EarliestTransformTime);
+
+                Assert.AreEqual(3000, (background.Elements[0] as StoryboardAnimation)?.GetEndTime());
+                Assert.AreEqual(12000, (background.Elements[0] as StoryboardAnimation)?.EndTimeForDisplay);
```

Comment:
I'm confused. This is fine and all, but the only thing that reads `EndTimeForDisplay` is... this test? How is this having any discernible effect anywhere in game? In fact, checking the beatmap linked in the issue thread, this PR doesn't seem to fix anything, visually speaking.

From the xmldoc I presume the property was also meant to be used here:

https://github.com/ppy/osu/blob/14af2063e5ec2abb16f751480f1702f54b6fe9ad/osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs#L85

and maybe here?

https://github.com/ppy/osu/blob/14af2063e5ec2abb16f751480f1702f54b6fe9ad/osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs#L88

Reply:
Seems to be a case of commit staging failure. I did test it to fix with the lines you propose, yes.